module/VuFind/src/VuFind/Db/Service/UserListService.php

```diff
@@ -295,16 +295,22 @@ public function getListsContainingResource(
     }
 
     /**
-     * Get public lists other than the one in array.
+     * Get public lists.
      *
-     * @param array $lists List of public lists.
+     * @param array $lists Ids or objects of public lists.
      *
      * @return array
      */
     public function getPublicLists($lists)
     {
-        $dql = 'SELECT ul FROM ' . $this->getEntityClass(UserList::class) . ' ul '
-            . 'WHERE ul NOT IN (:lists) AND ul.public = 1 ';
+        $dql = 'SELECT ul FROM ' . $this->getEntityClass(UserList::class) . ' ul ';
+
+        if (!empty($lists) && is_object($lists[0])) {
+            $dql .= 'WHERE ul.id IN (:lists) AND ul.public = 1';
+        } else {
+            $dql .= 'WHERE ul NOT IN (:lists) AND ul.public = 1';
```

Comment:
In the getLists method of ListItems.php public lists are fetched based on a condition and array merge is performed. It can also be done in another way just by modifing the where clause to just by setting ul.public. Either way this query is executed twice based on the condition in getLists method.

Reply:
@skellamp, my point of concern is that if we receive an array of objects, we perform an `IN` query, but if we receive an array of identifiers, we perform a `NOT IN` query. This seems unintentional, unless I'm misunderstanding something!

Reply:
That's a valid concern, would you prefer passing in a boolean selection parameter?

Reply:
Do we ever actually need to do a NOT IN query? I'm not sure what the circumstances would be where we would want to load every list EXCEPT those specified. I would think normally a plain IN would be desired. But if there's a use case for both scenarios, then adding a parameter to control that behavior independently of the format of the input data would make sense.

Reply:
Or is this about finding additional public lists? I guess that would make sense... and should definitely be explicitly requested.

Reply:
I modified both methods, the case in which we get additional public lists and merging them with baseLists is equal to getting all the public lists. May be the comments need to be modified accordingly.